### DISCUSSION TITLE
Any way to create a nice choice prompt?

### DISCUSSION THREAD

**Discussion by Jamalam360** [ajalt/clikt#509] [Q&A] [answered]
Is there any way to create an option that prompts with a list of options, allowing the user to use the arrow keys to move between the options? So far I've only found `option.choice(options).prompt(...)` which doesn't quite achieve this

Here's a screenshot from Inquirer.js of what I am talking about:

![image](https://github.com/ajalt/clikt/assets/56727311/3f5456bf-2daa-4617-89fd-eb6832356db9)

Thanks

**Comment by Jamalam360** [comment]
I made my own Mordant `Prompt` subclass (had to set the terminal to raw mode to capture arrow keys). Is there a way to use my custom prompt class in `option().choice(...).prompt()`?

**Reply by ajalt** [reply, marked as answer]
You can do that with `transformAll` with something like like `option().choice().transformAll { it.lastOrNull() ?: MyPrompt().ask() }`

**Reply by Jamalam360** [reply]
Thanks!
